client/AI/AIClientApp.cpp

```diff
@@ -96,22 +112,21 @@ void AIClientApp::Run() {
         ErrorLogger() << "AIClientApp::Run : Failed to connect to localhost server after " << MAX_TRIES << " tries.  Exiting.";
         Exit(1);
     }
+}
 
-    // join game
-    Networking().SendMessage(JoinGameMessage(PlayerName(), Networking::CLIENT_TYPE_AI_PLAYER));
-
-    // respond to messages until disconnected
-    while (1) {
-        if (!Networking().Connected())
-            break;
-        if (Networking().MessageAvailable()) {
-            Message msg;
-            Networking().GetMessage(msg);
-            HandleMessage(msg);
-        } else {
-            boost::this_thread::sleep_for(boost::chrono::milliseconds(250));
-        }
+void AIClientApp::StartPythonAI() {
+    PythonAI* python_ai = new PythonAI();
+    if (!(python_ai->Initialize())) {
+        const std::string err_msg = "AIClientApp::AIClientApp : Failed to initialize Python AI.  Exiting Soon.";
+        ErrorLogger() << err_msg;
+        Networking().SendMessage(
+            ErrorMessage(str(FlexibleFormat(UserString("ERROR_PYTHON_AI_CRASHED")) % PlayerName()) , true));
```

Comment:
At this point in the initialization, before the AI has tried to join the game the server's PlayerConnection is not initialized with the AI id or name.  It gets initialized when the AI joins.  I added a comment to clarify and left the code as is.